File: Src/Icm.Core.Tests/Nullable2/Nullable2Test.vb
Imports Icm

<TestFixture(), Category("Icm")>
Public Class Nullable2Test

    <Test()>
    Public Sub HasValueTest()
        Dim nulInteger As Nullable2(Of Integer)
        Assert.IsFalse(nulInteger.HasValue)

        nulInteger = Nothing
        Assert.IsFalse(nulInteger.HasValue)

        nulInteger = 0
        Assert.That(nulInteger.HasValue)

        nulInteger = Nothing
        Assert.IsFalse(nulInteger.HasValue)

        Dim nulString As Nullable2(Of String) = Nothing

        Assert.That(nulString.HasValue)
        nulString = Nothing
        Assert.That(nulString.HasValue)
        nulString = ""
        Assert.That(nulString.HasValue)
        nulString = "Abc"
        Assert.That(nulString.HasValue)

    End Sub

    <Test()>
    Public Sub ValueTest()
        Dim nulInteger As Nullable2(Of Integer)

        nulInteger = 0
        Assert.DoesNotThrow(
            Sub()
                Dim target = nulInteger.Value
            End Sub
        )

        nulInteger = Nothing
        Assert.Throws(Of InvalidOperationException)(
            Sub()
                Dim target = nulInteger.Value
            End Sub
        )

        Dim nulString As Nullable2(Of String) = Nothing

        Assert.DoesNotThrow(
            Sub()
                Dim target = nulString.Value
            End Sub
        )

        nulString = "Abc"
        Assert.DoesNotThrow(
            Sub()
                Dim target = nulString.Value
            End Sub
        )

    End Sub

    <Test()>
    Public Sub HasSomethingTest()
        Dim nulInteger As Nullable2(Of Integer)
        Assert.IsFalse(nulInteger.HasSomething)

        nulInteger = Nothing
        Assert.IsFalse(nulInteger.HasSomething)

        nulInteger = 0
        Assert.That(nulInteger.HasSomething)

        nulInteger = Nothing
        Assert.IsFalse(nulInteger.HasSomething)

        Dim nulString As Nullable2(Of String) = Nothing

        Assert.IsFalse(nulString.HasSomething)
        nulString = Nothing
        Assert.IsFalse(nulString.HasSomething)
        nulString = ""
        Assert.That(nulString.HasSomething)
        nulString = "Abc"
        Assert.That(nulString.HasSomething)

    End Sub

End Class

File: Src/Icm.Core.Tests/Nullable2/Nullable2Test.cs

using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using Icm;
using NUnit.Framework;

[TestFixture(), Category("Icm")]
public class Nullable2Test
{

	[Test()]
	public void HasValueTest()
	{
		Nullable2<int> nulInteger = default(Nullable2<int>);
		Assert.IsFalse(nulInteger.HasValue);

		nulInteger = null;
		Assert.IsFalse(nulInteger.HasValue);

		nulInteger = 0;
		Assert.That(nulInteger.HasValue);

		nulInteger = null;
		Assert.IsFalse(nulInteger.HasValue);

		Nullable2<string> nulString = null;

		Assert.That(nulString.HasValue);
		nulString = null;
		Assert.That(nulString.HasValue);
		nulString = "";
		Assert.That(nulString.HasValue);
		nulString = "Abc";
		Assert.That(nulString.HasValue);

	}

	[Test()]
	public void ValueTest()
	{
		Nullable2<int> nulInteger = default(Nullable2<int>);

		nulInteger = 0;
		Assert.DoesNotThrow(() =>
		{
			dynamic target = nulInteger.Value;
		});

		nulInteger = null;
		Assert.Throws<InvalidOperationException>(() =>
		{
			dynamic target = nulInteger.Value;
		});

		Nullable2<string> nulString = null;

		Assert.DoesNotThrow(() =>
		{
			dynamic target = nulString.Value;
		});

		nulString = "Abc";
		Assert.DoesNotThrow(() =>
		{
			dynamic target = nulString.Value;
		});

	}

	[Test()]
	public void HasSomethingTest()
	{
		Nullable2<int> nulInteger = default(Nullable2<int>);
		Assert.IsFalse(nulInteger.HasSomething);

		nulInteger = null;
		Assert.IsFalse(nulInteger.HasSomething);

		nulInteger = 0;
		Assert.That(nulInteger.HasSomething);

		nulInteger = null;
		Assert.IsFalse(nulInteger.HasSomething);

		Nullable2<string> nulString = null;

		Assert.IsFalse(nulString.HasSomething);
		nulString = null;
		Assert.IsFalse(nulString.HasSomething);
		nulString = "";
		Assert.That(nulString.HasSomething);
		nulString = "Abc";
		Assert.That(nulString.HasSomething);

	}

}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
